### PR TITLE
add link to filter attendances

### PR DESCRIPTION
**Pull request by jshawl**
To make taking attendance easier, there is a now a link that allows the user to filter attendances where the status is nil.

At first, I thought about doing this in JS, but the url params will allow us to send the filter link to others.

### CODE REVIEW

File: app/controllers/events_controller.rb [hunk 1]
```diff
@@ -19,7 +19,11 @@ def show
     @event = Event.find(params[:id])
     @current_user_is_admin = @event.group.has_admin?(current_user)
     redirect_to(current_user, flash:{alert: "You're not authorized."}) if !@current_user_is_admin
-    @attendances = @event.attendances
+    if params[:filter] == "na"
```
Comment (jshawl) [inline app/controllers/events_controller.rb]: feels a little smelly. myabe `params[:status] == "nil"`?

File: app/views/events/show.html.erb [hunk 2]
```diff
@@ -1,4 +1,14 @@
 <h2><%= "#{@event.title}, #{@event.date_s}" %></h2>
+
+
+<p>
+<% if request.url.match(/filter=na/) %>
+  <%=link_to "Show All", request.url.gsub(/filter=na/,'') %>
+<% else %>
+  <%=link_to "Filter by N/A", request.url + "&filter=na" %>
```
Comment (jshawl) [inline app/views/events/show.html.erb]: will break if there isn't a `?` before

File: app/controllers/events_controller.rb (hunk 1 above)
Comment (RobertAKARobin) [inline app/controllers/events_controller.rb]: Yeah, I'd prefer `params[:status]`. May as well match the name of the field.

Comment (RobertAKARobin) [inline app/controllers/events_controller.rb]: Also, I'd always forget whether `?filter=na` filters *on* `na` or filters *out* `na`.

File: app/views/events/show.html.erb [hunk 3]
```diff
@@ -1,4 +1,14 @@
 <h2><%= "#{@event.title}, #{@event.date_s}" %></h2>
+
+
+<p>
+<% if request.url.match(/filter=na/) %>
```
Comment (RobertAKARobin) [inline app/views/events/show.html.erb]: Use `params[:status]` here? RegEx seems a lil clunky

File: app/views/events/show.html.erb (hunk 2 above)
Comment (RobertAKARobin) [inline app/views/events/show.html.erb]: How about `link_to "Filter by N/A", event_path(event, status: "na")`? http://stackoverflow.com/a/2696001/2053389

**Comment by jshawl** [comment]
@RobertAKARobin whachu think now? https://github.com/ga-dc/garnet/commit/d1cdefd105bf38cb5ee9c9303bbacaf7cb799a27

File: app/views/events/show.html.erb [hunk 4]
```diff
@@ -1,4 +1,14 @@
 <h2><%= "#{@event.title}, #{@event.date_s}" %></h2>
+
+
+<p>
+<% if params[:status] == "nil" %>
+  <%=link_to "Show All", request.url.gsub(/&status=nil/,'') %>
+<% else %>
+  <%=link_to "Filter by N/A", request.url + "&status=nil" %>
```
Comment (RobertAKARobin) [inline app/views/events/show.html.erb]: How about `link_to "Filter by N/A", event_path(event, status: "na")`? `link_to` has a built-in querystring-adder.  http://stackoverflow.com/a/2696001/2053389

Comment (jshawl) [inline app/views/events/show.html.erb]: effff yteaaaah just wat i was lookin fo

**Pull request merged**